doc/developer/index.adoc

```diff
@@ -26,29 +26,136 @@ These conditions are to ensure that existing user code will continue to work whe
 
 For patch releases we never allow changes in the public API, unless the change is specifically to fix a bug that aligns the actual behavior of the code with the publicly documented behavior.
 
+Only release branches (more on those later) include the full major.minor.patch
+version numbers included with releases. The major and minor version of the _master_ and
+_develop_ branches do not include a patch number and always have a SNAPSHOT tag.
+The _master_ and _develop_ versions always matches the pattern *MAJOR*.*MINOR*-SNAPSHOT.
+
+The _master_ version always has the same major and minor number as the latest
+release while the _develop_ version uses the next expected major/minor
```

Comment:
If master has the same major/minor number as the latest release, that will imply in maven terms that it is prior to the last release, rather than after it. I would prefer to follow a principle of least surprise and increment the patch version on master after each release, rather than clearing the patch version and defaulting back to not having it.